src/CoreWCF.Http/src/CoreWCF/Channels/RequestDelegateHandler.cs

```diff
@@ -97,6 +98,21 @@ private WebSocketOptions CreateWebSocketOptions(HttpTransportBindingElement tbe)
 
         internal async Task HandleRequest(HttpContext context)
         {
+            if (SecurityUtils.ShouldUseAuthentication(_serviceDispatcher.Binding))
+            {
+                string scheme = _httpSettings.AuthenticationScheme.ToString();
+                AuthenticateResult authenticateResult = await context.AuthenticateAsync(scheme);
+                if (authenticateResult.None)
+                {
+                    await context.ChallengeAsync(scheme);
+                    return;
+                }
+                else
+                {
+                    context.Features.Set(authenticateResult);
+                }
```

Comment:
Features isn't the best place to stash this. Features is really about "do this to the request for me". You definitely shouldn't be putting a class in there that you don't own the type for. It's intended for middleware to put things in there for end user service implementations to use. There's an Items dictionary on HttpContext if you want to stash things. Use a CoreWCF specific name so there's no risk of name clashes. But I don't think you even need to stash this anyway. I'll comment more when I get to it.